airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationIntegrationTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.snowflake;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.string.Strings;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class SnowflakeDestinationIntegrationTest {
+  private final SnowflakeSQLNameTransformer namingResolver = new SnowflakeSQLNameTransformer();
+
+  @Test
+  public void testInvalidSchemaName() throws IOException {
+    final JsonNode config = Jsons.deserialize(new String(Files.readAllBytes(Paths.get("secrets/insert_config.json"))));
+    final String schemaName = "schemaName with whitespace "+ Strings.addRandomSuffix("integration_test", "_", 5);
+    ((ObjectNode) config).put("schema", schemaName);
+    assertThrows(SQLException.class, () -> syncWithoutNamingResolver(config));
```

Comment:
I don't think this is actually testing the `SnowflakeDatabase` class at all? I think it would make more sense to call `SnowflakeDatabase.getConnection(config)`, execute the `createSchemaQuery`, and validate that it does _not_ throw an exception.

That way, if anyone accidentally removes the `nameTransformer.getIdentifier` call, it would cause this test to fail.

Reply:
created new method to call SnowflakeDatabase.getConnection(config) to cover both test cases - with failing connection and without